heft-plugins/heft-storybook-plugin/src/schemas/storybook.schema.json

```diff
@@ -28,6 +28,11 @@
       "description": "If this is empty, then it will use the storybook default output dir. If you want to change the static build output dir to staticBuildDir, then the static build output dir would be: `\"staticBuildOutputFolder\": \"staticBuildDir\"`",
       "type": "string",
       "pattern": "[^\\\\]"
+    },
+    "storybookPackageNameTarget": {
+      "title": "Specifies an NPM dependency name that is used as the (cwd) target for the storybook commands",
+      "description": "By default the plugin executes the storybook commands in the local package context, but for distribution purposes it can be useful to split the TS library and storybook exports into two packages. For example, If you create an 'empty' storybook 'app' project for distribution purposes, then the storybook package name would be: `\"storybookPackageName\": \"my-ui-storybook-library\"`",
+      "type": "string"
     }
```

Comment:
I'm also curious about the importance of continuing to support Storybook 6.x with Heft, since version 7 is now almost a year old.

As a quick test, I added up the 7-day download counts for versions of [@storybook/react](https://www.npmjs.com/package/@storybook/react?activeTab=versions) over the past week (using old fashioned RegEx+Excel, since GPT choked on this problem 😝):

- **6.x versions:** 1.37M downloads
- **7.x version:** 2.6M downloads

Not sure how accurate this is, but it does seem that 6.x is still somewhat popular?

Reply:
I would recommend backwards compatibility

Not because of the potential challenges migrating to storybook 7, but because all heft-plugins have a strict peer dependency on the latest heft version. 

This means that if one upgrades heft, one is forced to upgrade all the heft plugins. And if those enforce larger breaking changes (like storybook 7) it can become rather complex to upgrade. As those could require not just tool but also source code change.

To support an incremental migration path for users using heft plugins, we should ideally loosen one of the peer dependencies being heft or the wrapping tool (like SB). 

Considering the impact of supporting both SB 6.x and 7.x seems minimal from the plugin perspective, I believe this to be the recommended path.